### PR TITLE
detect: Improve support for larger address values.

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of draft pr #5498 

This PR modifies the address rule value parsing to permit larger address values in the rule variables section.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [3887](https://redmine.openinfosecfoundation.org/issues/3887)

Describe changes:
- Set the size of the container to hold the address being parsed using this value or a default.
- Add unit test case to ensure maximum value is honored.

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

#suricata-verify-pr:
#suricata-verify-repo:
#suricata-verify-branch:
#suricata-update-pr:
#suricata-update-repo:
#suricata-update-branch:
#libhtp-pr:
#libhtp-repo:
#libhtp-branch:

### CODE REVIEW

File: src/detect-engine-address.c [hunk 1]
```diff
@@ -733,17 +733,16 @@ static int DetectAddressSetup(DetectAddressHead *gh, const char *s)
...
+        DetectAddressHead *ghn, const char *s, int negate, ResolvedVariablesList *var_list,
         int recur)
 {
     size_t x = 0;
     size_t u = 0;
     int o_set = 0, n_set = 0, d_set = 0;
     int depth = 0;
     size_t size = strlen(s);
-    char address[8196] = "";
+    char address[size < 512 ? 512 : size];
```
Comment (catenacyber) [inline src/detect-engine-address.c:745]: Should we rather alloc it on the heap than the stack ?

Comment (victorjulien) [reply src/detect-engine-address.c:745]: I think so. IIRC we can take this directly from the yaml, which means the size could be very large and exhaust stack?

**Comment by jlucovsky** [comment]
Continued in #5602
